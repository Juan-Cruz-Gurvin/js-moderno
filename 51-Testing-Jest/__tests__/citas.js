import Citas from "../js/classes/Citas";


describe('Probar la clase de citas', () => {

    const citas = new Citas();

    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Felipe',
            propietario: 'Juan',
            telefono: '3232423',
            fecha: '08-08-2022',
            hora:'10:30',
            sintomas: 'vacunas'
        };

        citas.agregarCita(citaObj);

        // Prueba 
        expect(citas).toMatchSnapshot();
    });

    test('actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Nativa',
            propietario: 'Juan',
            telefono: '3232423',
            fecha: '08-08-2022',
            hora:'10:30',
            sintomas: 'vacunas'
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test('Eliminar Cita', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    });
});