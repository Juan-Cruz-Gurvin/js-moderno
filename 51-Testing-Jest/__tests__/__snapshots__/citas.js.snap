// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "08-08-2022",
      "hora": "10:30",
      "id": 1660001108997,
      "mascota": "Felipe",
      "propietario": "Juan",
      "sintomas": "vacunas",
      "telefono": "3232423",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar Cita 1`] = `
Citas {
  "citas": Array [],
}
`;

exports[`Probar la clase de citas actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "08-08-2022",
      "hora": "10:30",
      "id": 1660001108997,
      "mascota": "Nativa",
      "propietario": "Juan",
      "sintomas": "vacunas",
      "telefono": "3232423",
    },
  ],
}
`;
